### PR TITLE
drivers/xbee: misc fixes 

### PR DESCRIPTION
**Pull request by haukepetersen**
- fixed sending with long addresses
 - adapted default values for channel and PAN (to be the same as the Atmel radio driver)
 - mark netif header pktsnip now as type `NG_NETTYPE_NETIF`

### CODE REVIEW

File: drivers/include/xbee.h [hunk 1]
```diff
@@ -67,12 +67,12 @@
 /**
  * @brief   Default PAN ID used after initialization
  */
-#define XBEE_DEFAULT_PANID          (0x0001)
+#define XBEE_DEFAULT_PANID          (0x0023)
```
Comment (authmillenon) [inline drivers/include/xbee.h]: Isn't this pretty arbitrary?

File: drivers/include/xbee.h [hunk 2]
```diff
@@ -67,12 +67,12 @@
...
 
 /**
  * @brief   Default channel used after initialization
  */
-#define XBEE_DEFAULT_CHANNEL        (11U)
+#define XBEE_DEFAULT_CHANNEL        (17U)
```
Comment (authmillenon) [inline drivers/include/xbee.h]: see https://github.com/RIOT-OS/RIOT/pull/2836/files#r28734438

**Comment by FrancescoErmini** [comment]
Channel and PAN ID must be the same for all devices in the network, so in my opinion it's ok to  have those values defined with macros. Though I argue that the Xbee manufactory value are in hex
0x0C for channel and 0x3332 for PAN ID, but those are only details, for me it's ok.

**Comment by authmillenon** [comment]
Yes of course, but why not use the already defined ones and adopt newer drivers?

**Comment by haukepetersen** [comment]
Yupp, the (default) values above are arbitrary. I think the most important thing here is, that all RIOT devices use the same default values (which I made sure through this PR). And I don't like to use the first value in the possible value range, but some arbitrary value in the middle. This makes sure, that 'broken' drivers do not hit these values accidentally and one thinks that their driver implementations work - even when they are not...

**Comment by authmillenon** [comment]
Okay, your argument is valid. ACK.

**Comment by kaspar030** [comment]
Linux setd default pan to to 0xffff. IMHO that makes sense. What do you think?

**Pull request merged**